5-appinfra/modules/env_baseline/db.tf

```diff
@@ -12,44 +12,88 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# CloudSQL Postgres instance
-module "cloudsql" {
-  source  = "GoogleCloudPlatform/sql-db/google//modules/postgresql"
-  version = "~> 20.0"
-
-  for_each = toset(var.cluster_regions)
-
-  project_id = var.app_project_id
-  region     = each.value
-
-  name                = "${var.db_name}-${each.value}-${var.env}"
-  database_version    = "POSTGRES_14"
-  enable_default_db   = false
-  tier                = "db-custom-1-3840"
-  deletion_protection = false
-  availability_type   = "REGIONAL"
-
-  ip_configuration = {
-    ipv4_enabled                  = false
-    psc_enabled                   = true
-    psc_allowed_consumer_projects = [var.cluster_project_id]
+locals {
+  region = var.cluster_regions[0]
+}
+
+# Create alloydb cluster and instance.
+
+data "google_project" "project" {
+  project_id = var.network_project_id
+}
+
+module "alloydb" {
```

Comment:
Looks like the ledger-db tf was removed, is this single instance now being used for both "accounts-db" and "ledger-db"?  Assuming the decision is to move to a single DB, seems like it shouldn't be created under accounts-userservice, but maybe a new shared db path?

Reply:
it looks alloydb asks user to use psql to create DB, unlike cloudsql, which can create DB in Terraform when creating the instance. as a result of this, I think we can just create one instance. Yes, it's better to use a shared DB path. @gtsorbo what do you think?

Reply:
Here is the architecture diagram, looks like the intention is a separate project and alloydb instance for each of the accounts and ledger databases.

https://cloud.google.com/architecture/enterprise-application-blueprint/cymbal-bank

Reply:
as discussed, added the same to ledger, so we have two, one for accounts, one for ledger.